tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsAggregatorTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿// <copyright file="DataStreamsAggregatorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DataStreamsMonitoring.Aggregation;
+using Datadog.Trace.DataStreamsMonitoring.Hashes;
+using Datadog.Trace.DataStreamsMonitoring.Utils;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Vendors.Datadog.Sketches;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+
+public class DataStreamsAggregatorTests
+{
+    private const long OneSecondNs = 1_000_000_000;
+    private const int BucketDurationMs = DataStreamsConstants.DefaultBucketDurationMs; // 10s
+    private const long BucketDurationNs = ((long)BucketDurationMs) * 1_000_000;
+
+    private static readonly long T1 = DateTimeOffset.UtcNow.ToUnixTimeNanoseconds();
+
+    // Set tp2 to be some 40 seconds after the tp1, but also account for bucket alignments,
+    // otherwise the possible StatsPayload would change depending on when the test is run.
+    private static readonly long T2 = BucketStartTimeForTimestamp(T1 + (40 * OneSecondNs)) + (6 * OneSecondNs);
+
+    // Based on the go aggregator_tests
+    // https://cs.github.com/DataDog/data-streams-go/blob/6772b163707c0a8ecc8c9a3b28e0dab7e0cf58d4/datastreams/aggregator_test.go#L28
+
+    [Fact]
+    public void Aggregator_DoesNotFlushCurrentStats()
+    {
+        var aggregator = CreateAggregatorWithData(T1, T2);
+
+        // flush at t2 doesn't flush points at t2 (current bucket)
+        // so only the last point is included
+        var statsToWrite = aggregator.Export(T2);
+
+        // compare expected, bit messy, but works
+        var stats = statsToWrite[0];
```

Comment:
Maybe you could do the following so you're not dependent on the order of the stats:

```suggestion
        var currentStats = stats.Where(x => x.TimestampType == TimestampType.Current);
        currentStats.Should().ContainSingle();
```

Reply:
Good call, thanks 👍 done in https://github.com/DataDog/dd-trace-dotnet/pull/3191/commits/6663e4ce707a16ebef2658f6f55147e8735afaf7